src/Registry.php

```diff
@@ -40,18 +40,23 @@ class Registry implements \JsonSerializable, \ArrayAccess, \IteratorAggregate, \
      *
      * @var    string
      * @since  1.4.0
+     *
+     * @deprecated This property will become protected in version 4 to prevent direct access.
```

Comment:
For now make it protected and add a magic setter and getter methods that trigger deprecations in code. Easier for logging than these ones in doc blocks

Reply:
Like how we dealt with WebApplication->$input in the application package https://github.com/joomla-framework/application/blob/2.0-dev/src/AbstractWebApplication.php#L228-L238 (obviously there we only added read access - here we'd add set and get

Reply:
seems like a good idea.

Reply:
@Fedik making the separator protected, and adding the magic set and get will help making it clear and will not break anything. So will you please go head and do that? Please make sure to follow this example from the AbstractWebApplication, for both, so that we don't allow any other access to the class properties.

Reply:
I can add magic. But magic get/set, changes the behavior of object. This can have a side efects.
Example if the code creating a property "on fly" it may crash the App in unexpected way. Like:
```php
class FooBar{
  public function __get($name){ return null; }
}

$c = new FooBar;
$c->potato[1] = 'foobar';

var_dump($c->potato[1] ); // Will be NULL instead of "foobar"
```

Not like it correct way to do the thing, but PHP allows it for now :wink: 

Another side effect: when the child class (which extend Registry) have own magic methods, and it will not call parrent magic, then it also will not work.

If you guys fine with it, I will add get/set.
I did not seen any of this in CMS, I guess all should be fine.

Reply:
In the magic getter/setter, you only accept 'separator' as a valid value. If the name is 'separator', get/set the value and issue a deprecation warning/notice, otherwise, throw an exception.